src/index.test.ts

```diff
@@ -3,12 +3,12 @@ import exported from '.'
 
 test('export', (t): void => {
   const expected = {
-    parser: '@typescript-eslint/parser',
     extends: 'eslint-config-standard',
     plugins: ['@typescript-eslint'],
     overrides: [
       {
         files: ['*.ts', '*.tsx'],
+        parser: '@typescript-eslint/parser',
```

Comment:
TypeScript makes possible many rules that require type information. Some such rules were already written in `@typescript-eslint/eslint-plugin` and many potential ones are yet to be written.

This change means that JavaScript files will no longer enjoy any of those rules. Would you agree?

While in many TypeScript projects the only JavaScript files are configuration files for various tools, some projects are mixed TS/JS where the JS could benefit from the TS rules, I suppose.

but... already these rules are under an override of `['*.ts', '*.tsx']`, which means only the TypeScript files enjoy them.

Perhaps optimal would be to allow the user to decide on which files to apply our shared configuration. Is it currently possible in ESLint to specify `extends` under `overrides`? Like:

```json
{
  "extends": "standard",
  "overrides": [
    {
      "files": ["src/**/*.ts"],
      "extends": "standard-with-typescript",
      "parserOptions": {
        "project": "./tsconfig.json"
      }
    }
  ]
}
```

Reply:
Right, this means `js` files can't benefit from the type-aware rules, but then again, this is already the case thanks to the overrides.

If somebody is using TypeScript for their project, the main reason to even have `js` files is to avoid the type system (like for configuration files). Otherwise, the user would have named the files `ts` in the first place.

There are some exceptions, like a legacy project that is slowly migrating to TypeScript module by module. Even in that case, applying type-aware rules to the `js` files would be undesirable, since those files are legacy code that still needs to be upgraded. If the user had the resources to fix those files, it would be simpler to just finish the port and change the file extensions, rather than to lint them halfway.

In other words, while your concern is valid, I also think it's also an extremely niche use-case. If you wanted to support this use-case, I suppose you could ship an `eslint-config-standard-with-typescript/includeJs` entry point in your library, so people can op-in to having the rules at the top level. For everyone else, the way it works today is pretty ideal (aside from this bug).

Reply:
And, for reference, this hypothetical `eslint-config-standard-with-typescript/includeJs` entry point might look something like this:

```js
const mainRules = require('./index.js')

module.exports = {
  ...mainRules.overrides[0]
}
```

So it's quick & easy to enable this other use case, without affecting usability for people who use the main `eslint-config-standard-with-typescript` entry point with its helpful overrides.